web_client/views/layout/HeaderImageView.js

```diff
@@ -33,9 +38,42 @@ var HeaderImageView = View.extend({
     render() {
         this.$el.html(headerImageTemplate({
             image: this.imageModel,
-            parentChain: this.parentChain
+            parentChain: this.parentChain,
+            nextImageLink: this.nextImageLink,
+            previousImageLink: this.previousImageLink
         }));
         return this;
+    },
+
+    _setNavigationLinks() {
+        const model = this.imageModel;
+        let analysisQuery = '';
+        if (!model) {
+            this.nextImageLink = null;
+            this.previousImageLink = null;
+            this.render();
+            return;
+        }
+
+        if (router.getQuery('analysis')) {
+            analysisQuery = `&analysis=${router.getQuery('analysis')}`;
+        }
+        $.when(
+            restRequest({
```

Comment:
Every time the analysis is changed, we refetch the next and previous images ids.  Similarly, when we first load an image with the analysis open we end up fetch the next and previous image ids twice.  Do we care about this inefficiency?  Should the next and previous ids only be fetch on h:imageOpened and the be reused?  I was testing in a folder with ~7000 images and this takes a noticeable time.

Reply:
Should be resolved now.  If performance of the endpoints becomes an issue, I can look into baking it into a mongo query rather than loading all of images into memory.